libexec/scoop-status.ps1

```diff
@@ -55,8 +55,10 @@ $true, $false | ForEach-Object { # local and global apps
     $dir = appsdir $global
     if (!(Test-Path $dir)) { return }
 
-    Get-ChildItem $dir | Where-Object name -NE 'scoop' | ForEach-Object {
-        $app = $_.name
+    [string[]] $appNames = Get-ChildItem $dir -Directory -Name -Exclude 'scoop' | Where-Object Length -GT 0
+    if ($appNames.Length -eq 0) { return }
+
+    foreach ($app in $appNames) {
```

Comment:
*   fix: refactor `Where-Object name -NE 'scoop'` to `Get-ChildItem`'s `Exclude` parameter. If the input object didn't have a `Name` parameter, an error would occur. This would happen when `Get-ChildItem` returns a zero-length array such as in the case of an empty or non-existent Global app dir.
*   fix: add `Where-Object Length -GT 0` to exclude zero-length strings from `$appNames`.
*   fix: assign app names array to `[string[]] $appNames`. Ensures the script halts if any other type is assigned to `$appNames` and prevents unexpected behavior caused by a type mismatch.
*   style: change `ForEach-Object` to `foreach`. A stylistic preference for the `$appName in $appNames` syntax over `$appName = $_`.